project-koku/koku
# [COST-4603] - Adding enhanced report status info

## Jira Ticket

[COST-4603](https://issues.redhat.com/browse/COST-4603)

## Description

This change will add new report and manifest statuses for tracking download/processing and summary tasks to completion

## Testing

1. Checkout Branch
2. Restart Koku
3. Make create/load test customer data
4. Everything should work correctly
5. You should also see some status checks in the db as follows:

Manifest entries:
state                         | {"summary": {"end": "2024-02-06T10:49:46.344037+00:00", "start": "2024-02-06T10:49:38.368181+00:00", "time_taken_seconds": 7}, "download": {"end": "2024-02-06T10:49:29.579968+00:00", "start": "2024-02-06T10
:49:27.977036+00:00", "time_taken_seconds": 1}, "processing": {"end": "2024-02-06T10:49:35.993112+00:00", "start": "2024-02-06T10:49:34.182368+00:00", "time_taken_seconds": 1}}

Report status entries:
celery_task_id          | 60db4099-ff13-4e4b-8285-48582392e7e8
failed_status           | 
status                  | 5

6. There is also so new entries on the source API as follow:

"download_status": "complete",
"processing_status": "complete",
"summary_status": "complete",

The above will cycle through `pending`, `in-progress` and `complete`

## Notes

...
